contract/r/gnoswap/emission/distribution.gno

```diff
@@ -0,0 +1,289 @@
+			))
+		}
+
+		pct := iPct.(uint64)
+		distAmount := calculateAmount(amount, uint64(pct))
+		totalSent += distAmount
+
+		transferToTarget(targetInt, distAmount)
+
+		return false
+	})
+
+	leftAmount := amount - totalSent
+	if leftGNSAmount > 0 {
+		setLeftGNSAmount(leftAmount)
+	}
+
+	return totalSent
+}
+
+// calculateAmount converts a basis point percentage to actual token amount
+// bptPct is in basis points (1/100th of 1%)
+// Example: 7500 basis points = 75%
+func calculateAmount(amount, bptPct uint64) uint64 {
+	return amount * bptPct / 10000
+}
+
+// transferToTarget sends tokens to the appropriate address based on target type
+// and updates both current and accumulated distribution tracking
+func transferToTarget(target int, amount uint64) {
+	switch target {
+	case LIQUIDITY_STAKER:
+		gns.Transfer(a2u(consts.STAKER_ADDR), amount)
+		distributedToStaker += amount
+		accuDistributedToStaker += amount
+
+	case DEVOPS:
+		gns.Transfer(a2u(consts.DEV_OPS), amount)
+		distributedToDevOps += amount
+		accuDistributedToDevOps += amount
+
+	case COMMUNITY_POOL:
+		gns.Transfer(a2u(consts.COMMUNITY_POOL_ADDR), amount)
+		distributedToCommunityPool += amount
+		accuDistributedToCommunityPool += amount
+
+	case GOV_STAKER:
+		gns.Transfer(a2u(consts.GOV_STAKER_ADDR), amount)
+		distributedToGovStaker += amount
+		accuDistributedToGovStaker += amount
+
+	default:
+		panic(addDetailToError(
+			errInvalidEmissionTarget,
+			ufmt.Sprintf("invalid target(%d)", target),
+		))
+	}
+}
+
+func GetDistributionBpsPct(target int) uint64 {
+	assertDistributionTarget(target)
+	iUint64, exist := distributionBpsPct.Get(strconv.Itoa(target))
+	if !exist {
+		panic(addDetailToError(
+			errInvalidEmissionTarget,
+			ufmt.Sprintf("invalid target(%d)", target),
+		))
+	}
+
+	return iUint64.(uint64)
+}
+
+func GetDistributedToStaker() uint64 {
+	return distributedToStaker
+}
+
+func GetDistributedToDevOps() uint64 {
+	return distributedToDevOps
+}
+
+func GetDistributedToCommunityPool() uint64 {
+	return distributedToCommunityPool
+}
+
+func GetDistributedToGovStaker() uint64 {
+	return distributedToGovStaker
+}
+
+func GetAccuDistributedToStaker() uint64 {
+	return accuDistributedToStaker
+}
+
+func GetAccuDistributedToDevOps() uint64 {
+	return accuDistributedToDevOps
+}
+
+func GetAccuDistributedToCommunityPool() uint64 {
+	return accuDistributedToCommunityPool
+}
+
+func GetAccuDistributedToGovStaker() uint64 {
+	return accuDistributedToGovStaker
+}
+
+func ClearDistributedToStaker() {
+	assertStakerOnly()
+	distributedToStaker = 0
+}
+
+func ClearDistributedToGovStaker() {
+	assertOnlyGovStaker()
+	distributedToGovStaker = 0
+}
+
+func setDistributionBpsPct(target int, pct uint64) {
+	if target == LIQUIDITY_STAKER {
+		oldPct, exist := distributionBpsPct.Get(strconv.Itoa(target))
+		if !exist {
+			panic("should not happen")
+		}
+
+		if oldPct.(uint64) != pct {
+			callbackStakerEmissionChange(calculateAmount(gns.GetEmission(), pct))
```

Comment:
second usage of `callbackStakerEmissionChange`, triggered by `ChangeDistributionPct` (called by admin or DAO).